flight/Modules/VtolPathFollower/vtol_follower_control.c

```diff
@@ -72,123 +197,184 @@ int32_t vtol_follower_control_path(const float dT, const PathDesiredData *pathDe
 {
 	PositionActualData positionActual;
 	PositionActualGet(&positionActual);
+
+	VelocityActualData velocityActual;
+	VelocityActualGet(&velocityActual);
+
+	const float cur_pos_ned[3] = {
+		positionActual.North +
+		    velocityActual.North*guidanceSettings.PositionFeedforward,
+		positionActual.East +
+		    velocityActual.East*guidanceSettings.PositionFeedforward,
+		positionActual.Down };
+
+	path_progress(pathDesired, cur_pos_ned, progress);
 	
-	const float cur[3] = { positionActual.North, positionActual.East, positionActual.Down };
-	
-	path_progress(pathDesired, cur, progress);
-	
+	// Interpolate desired velocity and altitude along the path
+	float groundspeed = vtol_interpolate(progress->fractional_progress,
+	    pathDesired->StartingVelocity, pathDesired->EndingVelocity);
+
+	if (progress->fractional_progress >= 1)
+		groundspeed = 0;
```

Comment:
I have the PR.  The falling back to endpoint following is basically the same--- or perhaps more aggressive, than the known groundspeed = 0 bug (because it's going to not just try and stop, but pull back).

I like the way you latch STATUS_COMPLETED; it's effectively level triggered.